utils/check-for-trailing-whitespace.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+cd .. || exit
+# print first
+grep -EHInr '( +)$' ./* | grep -v 'received: '
```

Comment:
It would be better to change the test code to avoid magic word `received: `.
Or is it acceptable to skip the check in `t/` file? @membphis

Reply:
> It would be better to change the test code to avoid the magic word `received: `.

agree with this. it seems that we have to update the test cases first.

we can create new issues about these to-do things, fix them later after this PR merged.